client/pages/expense/[eid]/index.js

```diff
@@ -1,16 +1,46 @@
 import Alert from 'react-bootstrap/Alert'
+import Button from 'react-bootstrap/Button'
+import { useState } from 'react'
+import { useRouter } from 'next/router'
 
-export default function Expense({ expense }) {
+export default function Expense({ url, expense }) {
+    const router = useRouter()
+    const [show, setShow] = useState(false)
+
+    function deleteExpense() {
```

Comment:
if this function can be async, i'd like to see it switch to async and use `await` instead of `.then()`